### PR TITLE
Release 1.5.2

### PR DESCRIPTION
**Pull request by benjamn**
[Meteor 1.5.1](https://github.com/meteor/meteor/pull/8788) had to be released sooner rather than later because of [a Node security update](https://github.com/meteor/meteor/issues/8896), but a lot of great work has been happening on `devel` that ought to be released before [Meteor 1.6](https://github.com/meteor/meteor/pull/8728), and that means we're going to need a follow-up Meteor 1.5.2 release.

### CODE REVIEW

**Comment by deadjdona** [comment]
any plans to update node.js to 4.8.4?
https://nodejs.org/en/blog/vulnerability/july-2017-security-releases/

**Comment by hwillson** [comment]
@deadjdona Already done and in 1.5.1 - see https://github.com/meteor/meteor/commit/afbe59d.

**Comment by sdarnell** [comment]
Probably related to changes in 1.5.2 (rather than the staringatlights:fast-render package), but with the Vue demo app at https://github.com/meteor-vue/vue-meteor-demo and meteor 1.5.2-beta.9 there's an exception in bindEnvironment() trying to access slice() in the dynamics:
```
 error on fast-rendering path: / ; error: TypeError: dynamics.slice is not a function
     at Object.Meteor.bindEnvironment (packages/meteor.js:1166:41)
     at PublishContext._.extend.onStop (packages/ddp-server/livedata_server.js:1226:23)
     at Context.processPublication (packages/staringatlights_fast-render.js:497:18)
...
```
Was fine on 1.5.2-beta.6, beta.7 but started on beta.8 (when meteor package went from 1.7.0 to 1.7.1.

**Comment by abernix** [comment]
@sdarnell There's already [an issue for this problem](https://github.com/abecks/meteor-fast-render/issues/4) (and [a pull-request which fixes the problem](https://github.com/abecks/meteor-fast-render/pull/5)) in the `staringatlights:fast-render` repository, however I've commented about the issue in https://github.com/meteor/meteor/issues/9003#issuecomment-321803612.

**Comment by sdarnell** [comment]
OK, great, thanks @abernix

**Comment by arggh** [comment]
Updating from `1.5.2-beta.8` to `1.5.2-beta.9` makes my app throw in a very generic situation, when navigating between two routes, both of which subscribe to a different subset of the same documents.

<details>
<summary>Stack</summary>
```

collection.js:160 Uncaught Error: Expected not to find a document already present for an add
    at Object.update (http://localhost:3000/packages/mongo.js?hash=1cd846c44ae3ffa5db19ced8d6d332eb2a3b764b:235:19)
    at Object.store.(anonymous function) [as update] (http://localhost:3000/packages/ddp-client.js?hash=14d966b7972bd95a1f7015fec9ac340f10508a44:3733:60)
    at http://localhost:3000/packages/ddp-client.js?hash=14d966b7972bd95a1f7015fec9ac340f10508a44:4540:19
    at Array.forEach (<anonymous>)
    at Function._.each._.forEach (http://localhost:3000/packages/underscore.js?hash=cde485f60699ff9aced3305f70189e39c665183c:149:11)
    at http://localhost:3000/packages/ddp-client.js?hash=14d966b7972bd95a1f7015fec9ac340f10508a44:4539:13
    at Function._.each._.forEach (http://localhost:3000/packages/underscore.js?hash=cde485f60699ff9aced3305f70189e39c665183c:157:22)
    at Connection._performWrites (http://localhost:3000/packages/ddp-client.js?hash=14d966b7972bd95a1f7015fec9ac340f10508a44:4535:9)
    at Connection._flushBufferedWrites (http://localhost:3000/packages/ddp-client.js?hash=14d966b7972bd95a1f7015fec9ac340f10508a44:4522:10)
    at Connection._livedata_data (http://localhost:3000/packages/ddp-client.js?hash=14d966b7972bd95a1f7015fec9ac340f10508a44:4488:12)
```

</details>

**Comment by benjamn** [comment]
@arggh Any chance you can put together a small reproduction for that bug?

**Comment by arggh** [comment]
@benjamn Sure thing, give me a second.

**Comment by arggh** [comment]
Here comes: https://github.com/arggh/meteor-1.5.2-beta-mongo-issue

### Steps to reproduce:

1. Run app
2. Click "Add doc" maybe three to four times
3. Click one of the links that appeared on the page
4. Click "Show all docs"

You should get an error like this thrown:

meteor-152-beta-screenshot

**Comment by benjamn** [comment]
@arggh My `git bisect` points to this commit: 2c5094fa0cdd7b7ae649a2d843cb73888eff0c4b

File: .circleci/config.yml [hunk 1]
```diff
@@ -124,9 +124,9 @@ jobs:
           name: Get Ready
           command: |
             eval $PRE_TEST_COMMANDS;
-            ./meteor --help
-          # shouldn't take longer than 5 minutes
-          no_output_timeout: 5m
+            ./meteor --get-ready
```
Comment (benjamn) [inline .circleci/config.yml]: Not only did this solve our testing problems (🤞), but it demonstrates another strength of @abernix's Circle 2.0 configuration: this command runs once, before any test jobs, so it only has one chance to fail. If it fails, it fails fast, and we can easily re-run the workflow.

**Comment by abernix** [comment]
Meteor 1.5.2 has officially moved into the Release Candidate phase!  It would be much appreciated if those who can could take the time to test now to help us move this toward an official, stable release.  As usual, this involves running:

    meteor update --release 1.5.2-rc.2

**Comment by filipenevola** [comment]
Hello, rc.2 is working fine in my app.

I just don't know why I get these warnings below because I checked those versions and they are updated on rc.2 source code.

I already ran meteor reset and I continue getting these warnings, should I clean some place else to have a build without any warning?

WARNING: Attempting to install plugin cordova-plugin-console@1.0.5, but it
         should have a minimum version of 1.0.7 to ensure compatibility with
         the current platform versions. Installing the minimum version for
         convenience, but you should adjust your dependencies.
WARNING: Attempting to install plugin cordova-plugin-statusbar@2.2.1, but it
         should have a minimum version of 2.2.3 to ensure compatibility with
         the current platform versions. Installing the minimum version for
         convenience, but you should adjust your dependencies.
WARNING: Attempting to install plugin cordova-plugin-splashscreen@4.0.1, but it
         should have a minimum version of 4.0.3 to ensure compatibility with
         the current platform versions. Installing the minimum version for
         convenience, but you should adjust your dependencies.

Anyway, I think we are ready to the final version of 1.5.2 :+1:

**Comment by abernix** [comment]
@filipenevola I was seeing those warnings when working on a 1.6 test project, but I haven't been able to figure out quite why.  I was able to confirm that the project was running with the correct plugins though, so the warnings do seem to just be that: warnings.

**Comment by abernix** [comment]
After far too many delays in releasing it, the final Meteor 1.5.2 release has been built and released. 🎉 🎊 🚀 

It's not marked as "recommended" yet (even though it is considered stable), which just means it won't install automatically yet.  Regardless, developers can update to it right now by running:

    meteor update --release 1.5.2

We will update it to be "recommended", most likely later today, but for now this PR is off to the `master` branch!

**Pull request merged**